From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <spottedleaf@spottedleaf.dev>
Date: Wed, 6 May 2020 15:56:55 -0700
Subject: [PATCH] Re-apply spigot's fix to not create regionfiles on read

Spigot commit: https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/b331a055090a1c5a25f6f03264b647fe956cf558

Paper accidentally reverted the `existingOnly` parameter to false,
which means the getFile function will never return null and will
always opt to create a regionfile.

diff --git a/src/main/java/net/minecraft/server/RegionFileCache.java b/src/main/java/net/minecraft/server/RegionFileCache.java
index ab07cfd4ef12aa2d25c10ca42ea767e6d2d0df69..9250b72ab4f9591dc5ac759d109d4299ec70641f 100644
--- a/src/main/java/net/minecraft/server/RegionFileCache.java
+++ b/src/main/java/net/minecraft/server/RegionFileCache.java
@@ -161,7 +161,7 @@ public class RegionFileCache implements AutoCloseable { // Paper - no final
     @Nullable
     public NBTTagCompound read(ChunkCoordIntPair chunkcoordintpair) throws IOException {
         // CraftBukkit start - SPIGOT-5680: There's no good reason to preemptively create files on read, save that for writing
-        RegionFile regionfile = this.getFile(chunkcoordintpair, false, true); // Paper
+        RegionFile regionfile = this.getFile(chunkcoordintpair, true, true); // Paper // Tuinity - revert paper oopsie
         if (regionfile == null) {
             return null;
         }
